### PR TITLE
Apollo graphql for tf

### PR DESCRIPTION
**Pull request by Akkikens**

### CODE REVIEW

File: src/app/TranscriptionFactor/[type]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import dynamic from "next/dynamic";
+
+const TranscriptionFactors = dynamic(
+  () => import("@/components/TranscriptionFactors")
+);
+const TranscriptionTabs = dynamic(
+  () => import("@/components/TranscriptionTabs")
+);
+
+type ContentType = "Human" | "Mouse";
```
Comment (NishiPhalke) [inline src/app/TranscriptionFactor/[type]/page.tsx:13]: Rename ContentType to species

File: src/app/page.tsx [hunk 2]
```diff
@@ -52,8 +52,7 @@ const Homepage = () => {
       iCRELdrQuery(study: $study) {
         snps
```
Comment (NishiPhalke) [inline src/app/page.tsx:53]: where do we need this query ?

File: src/components/CellType/CtPage.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+"use client";
+
+import React from "react";
+import { Box, Container, Typography } from "@mui/material";
+import Summary from "./Summary";
+
+const CtPage: React.FC<{ species: string }> = ({ species }) => {
+  const assembly = species === "human" ? "GRCh38" : "mm10";
```
Comment (NishiPhalke) [inline src/components/CellType/CtPage.tsx:8]: use one variable here

File: src/components/CellType/CtPage.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  const celltype = species === "human" ? "GRCh38" : "mm10";
+  const details = "summary";
```
Comment (NishiPhalke) [inline src/components/CellType/CtPage.tsx:10]: details not being used anywhere ?

File: src/components/CellType/Queries.tsx
Comment (NishiPhalke) [inline src/components/CellType/Queries.tsx:1]: This file can be named Queries.ts as we are not returning any TSX element here

File: src/components/CellType/Summary.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    },
+  });
+
+  if (loading) return <CircularProgress />;
+  if (error || ctError)
+    return <p>Error: {error?.message || ctError?.message}</p>;
+
+  // Adjusted to work with the correct structure
+  const rows = data?.peakDataset?.datasets ?? [];
+  console.log("rows:", rows);
```
Comment (NishiPhalke) [inline src/components/CellType/Summary.tsx:63]: remove console.logs

File: src/components/tf/Query.tsx [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+export const DATASETS_QUERY = gql(`
+  query Datasets1(
```
Comment (NishiPhalke) [inline src/components/tf/Query.tsx:40]: rename Datasets1 to something relevant

File: src/components/tf/Query.tsx [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+      }
+      uniprot_data
+      ncbi_data
+      factor_wiki
+    }
+  }
+`);
+
+export const TF_INFO_QUERY = gql(`
+  query Datasets2(
```
Comment (NishiPhalke) [inline src/components/tf/Query.tsx:154]: Rename this Datasets2 query as well

File: src/components/tf/TfDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+
+  const {
+    data: tfData,
+    loading: tfLoading,
+    error: tfError,
+  } = useQuery<TFInfoQueryResponse>(TF_INFO_QUERY, {
+    variables: {
+      processed_assembly: assembly,
+      replicated_peaks: true,
+      include_investigatedas: [
```
Comment (NishiPhalke) [inline src/components/tf/TfDetails.tsx:72]: use  includeTargetTypes from consts.ts here

File: src/components/tf/TfDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+        "cofactor",
+        "chromatin remodeler",
+        "RNA polymerase complex",
+        "DNA replication",
+        "DNA repair",
+        "cohesin",
+        "transcription factor",
+      ],
+      exclude_investigatedas: ["recombinant protein"],
```
Comment (NishiPhalke) [inline src/components/tf/TfDetails.tsx:81]: use  excludetargetTypes from consts.ts here

**Review by NishiPhalke — COMMENTED** [review]
Please use camelCase for variable names consistently throughout the code

**Review by NishiPhalke — APPROVED** [review]

File: src/components/tf/Query.tsx (hunk 6 above)
Comment (Akkikens) [reply src/components/tf/Query.tsx]: okay

**Pull request merged**